tests/test_cluster.go

```diff
@@ -335,9 +337,24 @@ func NewClusterWithPersistenceTestBaseFactory(t *testing.T, options *TestCluster
 }
 
 func setupIndex(esConfig *esclient.Config, logger log.Logger) error {
-	esClient, err := esclient.NewFunctionalTestsClient(esConfig, logger)
+	var esClient esclient.IntegrationTestsClient
+	op := func() error {
+		var err error
+		esClient, err = esclient.NewFunctionalTestsClient(esConfig, logger)
+		if err != nil {
+			return err
+		}
+
+		return esClient.Ping(context.TODO())
+	}
+
+	err := backoff.ThrottleRetry(
+		op,
+		backoff.NewExponentialRetryPolicy(time.Second).WithExpirationInterval(time.Minute),
+		nil,
+	)
 	if err != nil {
-		return err
+		logger.Fatal("Failed to connect to elasticsearch", tag.Error(err))
 	}
 
 	exists, err := esClient.IndexExists(context.Background(), esConfig.GetVisibilityIndex())
```

Comment:
Could you have done this without adding Ping, just by running IndexExists in a loop and retrying until it returns true/false with no error?